src/test/java/com/jwierzb/parkingspaces/controller/DriverControllerTest.java

```diff
@@ -0,0 +1,370 @@
+
+    @MockBean
+    UserDao userDao;
+    @MockBean
+    DriverTypeDao driverTypeDao;
+    @MockBean
+    RoleDao roleDao;
+    @MockBean
+    CurrencyDao currencyDao;
+    @MockBean
+    VehicleDao vehicleDao;
+    @MockBean
+    TransactionDao transactionDao;
+    @MockBean
+    JwtTokenProvider jwtTokenProvider;
+
+
+    @Test
+    @WithMockUser(username = "asdf", roles = "NOT_DRIVER_ROLE")
+    public void shouldForbidAccesToEndopintDueToLackOfRoles() throws Exception {
+        mockMvc.perform(get("/driver/vehicle/show")).andExpect(status().isForbidden());
+        mockMvc.perform(post("/driver/vehicle/register/test")).andExpect(status().isForbidden());
+        mockMvc.perform(delete("/driver/vehicle/delete/test")).andExpect(status().isForbidden());
+        mockMvc.perform(post("/driver/parking/test/start")).andExpect(status().isForbidden());
+        mockMvc.perform(put("/driver/parking/test/stop")).andExpect(status().isForbidden());
+        mockMvc.perform(get("/driver/parking/test")).andExpect(status().isForbidden());
+        mockMvc.perform(get("/driver/parking/history")).andExpect(status().isForbidden());
+
+    }
+
+
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void shouldAddVehicleToDriver() throws Exception {
+
+        String regNumber = "redddg";
+        UserEntity user = UserEntity.builder().username("user").build();
+        Vehicle vehicle = Vehicle.builder().id(0L).registrationNumber(regNumber.toUpperCase()).user(user).build();
+
+        when(vehicleDao.existsVehicleByUser(anyObject())).thenReturn(false);
+        when(vehicleDao.existsVehicleByRegistrationNumber(anyString())).thenReturn(false);
+        when(vehicleDao.save(anyObject())).thenReturn(vehicle);
+        when(userDao.findByUsername(anyString())).thenReturn(user);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(vehicle);
+
+        mockMvc.perform(post("/driver/vehicle/register/"+regNumber))
+                .andExpect(status().isOk())
+                .andExpect(content().string(jsonInString));
+
+    }
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void shouldntAddVehicleToDriverHavingVehicle() throws Exception {
+
+        String regNumber = "redddg";
+
+        when(vehicleDao.existsVehicleByUser(anyObject())).thenReturn(true);
+
+        mockMvc.perform(post("/driver/vehicle/register/"+regNumber))
+                .andExpect(status().isForbidden());
+
+    }
+    @Test
+    @WithMockUser(username = "test_user", roles = {"DRIVER"})
+    public void shouldntAddVehicleTwice() throws Exception {
+
+        String regNumber = "redddg";
+
+        //vehicle exists in db
+        when(vehicleDao.existsVehicleByRegistrationNumber(anyString())).thenReturn(true);
+
+        mockMvc.perform(post("/driver/vehicle/register/"+regNumber))
+                .andExpect(status().isForbidden());
```

Comment:
jak chcemy dodać drugi raz samochód to nie powinniśmy dostawać 403 - to nie jest kwestia uprawnień